src/main/java/com/epam/esm/persistence/jdbc/gift/JdbcTemplateGiftDao.java

```diff
@@ -0,0 +1,227 @@
+    private static final String SQL_WITH_NAME = " gift_certificate.name LIKE '%%%s%%'";
+    private static final String SQL_WITH_DESCRIPTION = " description LIKE '%%%s%%'";
+
+    private static final String SQL_FIND_GIFT_BY_ID = "SELECT gift_certificate.gift_id, " +
+            "gift_certificate.name, tag.name tag_name, " +
+            "gift_certificate.description,gift_certificate.price, " +
+            "gift_certificate.duration, gift_certificate.create_date, " +
+            "gift_certificate.last_update_date, gift_tags.gift_id, " +
+            "gift_tags.tag_id FROM gift_certificate " +
+            "LEFT JOIN gift_tags ON gift_tags.gift_id = gift_certificate.gift_id " +
+            "LEFT JOIN tag ON gift_tags.tag_id = tag.tag_id " +
+            "WHERE gift_certificate.gift_id = ?";
+
+    private static final String SQL_CREATE_GIFT = "INSERT INTO gift_certificate" +
+            "(name, description, price, duration, create_date, last_update_date) " +
+            "VALUES (?,?,?,?,?,?)";
+
+    private static final String SQL_UPDATE_GIFT = "UPDATE gift_certificate " +
+            "SET name = ?, description = ?, " +
+            "price = ?, duration = ?, last_update_date = ? " +
+            "WHERE gift_id = ?";
+
+    private static final String SQL_DELETE_GIFT = "DELETE FROM gift_certificate WHERE gift_id = ?";
+    private static final String SQL_INSERT_GIFT_TAGS = "INSERT INTO gift_tags(gift_id, tag_id) VALUES (?,?)";
+    private static final String SQL_DELETE_GIFT_TAGS = "DELETE FROM gift_tags WHERE gift_id = ?";
+
+    private static final String WHERE = " WHERE ";
+    private static final String AND = " AND ";
+    private static final String ORDER_BY = " ORDER BY";
+    private static final String GIFT_CERTIFICATE_NAME = " gift_certificate.name ";
+    private static final String COMMA = ", ";
+    private static final String CREATE_DATE = " create_date ";
+    private static final String NOT_FOUND = "Requested gift not found (id = %s)";
+
+    private final GiftMapperDB giftMapper;
+    private final JdbcTemplate jdbcTemplate;
+    private final BaseGiftValidator<GiftCertificate, Long> giftValidation;
+    private final JdbcTemplateTagDao jdbcTemplateTagDao;
+
+    @Autowired
+    public JdbcTemplateGiftDao(JdbcTemplate jdbcTemplate,
+                               BaseGiftValidator<GiftCertificate, Long> giftValidation, GiftMapperDB giftMapper, JdbcTemplateTagDao jdbcTemplateTagDao) {
+        this.jdbcTemplate = jdbcTemplate;
+        this.giftValidation = giftValidation;
+        this.giftMapper = giftMapper;
+        this.jdbcTemplateTagDao = jdbcTemplateTagDao;
+    }
+
+    @Override
+    public List<GiftCertificate> findAllEntities(GiftSearchFilter giftSearchFilter) {
+        giftValidation.onBeforeFindAllEntities(giftSearchFilter);
+        StringBuilder query = new StringBuilder(SQL_FIND_ALL_GIFTS);
+        StringBuilder queryParams = new StringBuilder();
+        StringBuilder orderParams = new StringBuilder();
```

Comment:
The method is huge. It's better to move at least the logic for building search requests somewhere else.